code/iaas/model/src/main/java/io/cattle/platform/core/dao/impl/InstanceDaoImpl.java

```diff
@@ -59,4 +62,27 @@ public boolean isOnSharedStorage(Instance instance) {
                 .fetchInto(InstanceRecord.class);
     }
 
+    @Override
+    public Instance getInstanceByExternalIdOrUuid(Long accountId, String externalId, String uuid) {
+        Condition accountCondition = INSTANCE.ACCOUNT_ID.eq(accountId).and(INSTANCE.REMOVED.isNull());
```

Comment:
This may be a convoluted situation but something to think about.  On the cattle side we look for external id first and then uuid.  On the python-agent side we look for uuid first and then external id.  Could this cause problems?  I'm not sure I can think of any, but just bringing it up.